handlers.go

```diff
@@ -1468,6 +1468,13 @@ func (s *Server) handleGetContentByCid(c echo.Context) error {
 		return err
 	}
 
+	//defer Cid panic handler
+	defer func() {
+		if err := recover(); err != nil {
+			log.Warnf("failed to get content by cid (%s)",obj.Hash(), err)
+		}
+	}()
+
 	v0 := cid.NewCidV0(obj.Hash())
```

Comment:
@alvin-reyes I have looked into this. The `cid` package panics for cidv0 that are not backward compatible e.g identity cids. Instead of constantly panicking and recovering, it's best to correctly handle cidv0 for both backward and forward compatibility.

I have tested this code locally and works just fine. Remove the defer func and replace line 1478 with;

```
v0 := cid.Undef
	dec, err := multihash.Decode(obj.Hash())
	if err == nil {
		if dec.Code == multihash.SHA2_256 || dec.Length == 32 {
			v0 = cid.NewCidV0(obj.Hash())
		}
	}
```

Reply:
@alvin-reyes you can verify with `bafkqaaa` before and after